### PR TITLE
 add st7735 lcd driver 

### PR DESCRIPTION
**Pull request by xfan1024**
support spi lcd screen st7735

### CODE REVIEW

File: components/drivers/spi/spi_lcd_st7735.c [hunk 1]
```diff
@@ -0,0 +1,368 @@
+#include <rtthread.h>
+#include <rtdevice.h>
+#include <rtgui/rtgui.h>
+#include <rtgui/driver.h>
+#include <rtgui/rtgui_server.h>
```
Comment (BernardXiong) [inline components/drivers/spi/spi_lcd_st7735.c]: Is it really related with GUI Engine?

**Comment by BernardXiong** [comment]
Any docs? Where is `Kconfig` options file?

File: components/drivers/spi/spi_lcd_st7735.c (hunk 1 above)
Comment (xfan1024) [reply components/drivers/spi/spi_lcd_st7735.c]: No, this is a misstake.

**Comment by xfan1024** [comment]
I never use Kconfig in my project, so I didn't write it. The driver have only one export function and I write doxygen-style comments on it, need any docs else?

**Pull request closed without merging**